dispatcher.go

```diff
@@ -65,10 +65,10 @@ type Config struct {
 	Inbounds  Inbounds
 	Outbounds Outbounds
 
-	// Filter and Interceptor that will be applied to all outgoing and incoming
-	// requests respectively.
-	Filter      transport.Filter
-	Interceptor transport.Interceptor
+	// Filters and Interceptors that will be applied to all outgoing and
+	// incoming requests respectively.
+	Filters      Filters
+	Interceptors Interceptors
```

Comment:
What's the motivation behind switching to plural here? I think we want singular cc @abhinav